pkg/scheduler/framework/plugins/default_registry.go

```diff
@@ -253,6 +257,22 @@ func NewDefaultConfigProducerRegistry() *ConfigProducerRegistry {
 			return
 		})
 
+	registry.RegisterPriority(requestedtocapacityratio.Name,
+		func(args ConfigProducerArgs) (plugins config.Plugins, pluginConfig []config.PluginConfig) {
+			plugins.Score = appendToPluginSet(plugins.Score, requestedtocapacityratio.Name, &args.Weight)
+			encoding, err := json.Marshal(args.RequestedToCapacityRatioArgs)
+			if err != nil {
+				klog.Fatalf("Failed to marshal %+v", args.RequestedToCapacityRatioArgs)
+				return
+			}
+			config := config.PluginConfig{
+				Name: requestedtocapacityratio.Name,
+				Args: runtime.Unknown{Raw: encoding},
+			}
```

Comment:
Can you factor this out into a private function, and also use it to replace lines 207-215